### PR TITLE
Minor code cleanup: ACLRule human_readable_repr as __unicode__ and smarter __repr__.

### PR DESCRIPTION
**Pull request by amcgregor**
Conversion to standard Python idiom of allowing the objects to be coerced to text.  Simplifies use in templates (`${foo}` calls `unicode()` anyway) and by having `__repr__` use the string representation of the object, saves a lot of (almost, but not quite) code duplication.

Also threw in a change to eliminate a silly `@staticmethod` if-tree.  (See the specific commit for additional notes on that change.)

### CODE REVIEW

**Comment by amcgregor** [comment]
Ugh; you might have to cherry pick to avoid the crazy merge up top.  :/  Sorry 'bout that.

File: brave/core/group/acl.py [hunk 1]
```diff
@@ -86,16 +86,15 @@ def __repr__(self):
                 self.ids
             )
 
-    def human_readable_repr(self):
-        return "{grant} if character is{not_}{prep} {set}".format(
+    def __unicode__(self):
+        return "{grant} if character {is}{prep} {set}".format(
                 grant='grant' if self.grant else 'deny',
-                not_=' not' if self.inverse else '',
+                not_='is not' if self.inverse else 'is',
```
Comment (jboning) [inline brave/core/group/acl.py:92]: is

File: brave/core/group/acl.py [hunk 2]
```diff
@@ -114,18 +113,11 @@ def evaluate(self, user, character):
...
-            )
-
-    def human_readable_repr(self):
-        return '{grant} if user has{not_} submitted a {kind} key'.format(
+    def __unicode__(self):
+        return '{grant} if user {has} submitted a {kind} key'.format(
                 grant='grant' if self.grant else 'deny',
-                not_=' not' if self.inverse else '',
-                kind=self.KINDS[self.kind]
+                not_='has not' if self.inverse else 'has',
```
Comment (jboning) [inline brave/core/group/acl.py:119]: has

**Comment by jboning** [comment]
Superseded by #369

**Pull request closed without merging**